docs/docs/aztec/concepts/smart_contracts/functions/inner_workings.md

```diff
@@ -94,3 +98,255 @@ This:
+
+This makes `#[aztec(view)]` functions suitable for critical read-only operations where the integrity of the result is crucial. Unconstrained functions, on the other hand, are executed entirely client-side without generating any proofs. It is better to use `#[aztec(view)]` if the result of the function will affect some sort of state, and they can be used for cross-contract calls.
+
+`#[aztec(view)]` functions can be combined with other Aztec attributes like `#[aztec(private)]` or `#[aztec(public)]`.
+
+## `Initializer` Functions #[aztec(initializer)]
+
+This is used to designate functions as initializers (or constructors) for an Aztec contract. These functions are responsible for setting up the initial state of the contract when it is first deployed. The macro does two important things:
+
+- `assert_initialization_matches_address_preimage(context)`: This checks that the arguments and sender to the initializer match the commitments from the address preimage
+- `mark_as_initialized(&mut context)`: This is called at the end of the function to emit the initialization nullifier, marking the contract as fully initialized and ensuring this function cannot be called again
+
+Key things to keep in mind:
+
+- A contract can have multiple initializer functions defined, but only one initializer function should be called for the lifetime of a contract instance
+- Other functions in the contract will have an initialization check inserted, ie they cannot be called until the contract is initialized, unless they are marked with [`#[aztec(noinitcheck)])`](#aztecnoinitcheck)
+
+## #[aztec(noinitcheck)]
+
+In normal circumstances, all functions in an Aztec contract (except initializers) have an initialization check inserted at the beginning of the function body. This check ensures that the contract has been initialized before any other function can be called. However, there may be scenarios where you want a function to be callable regardless of the contract's initialization state. This is when you would use `#[aztec(noinitcheck)]`.
+
+When a function is annotated with `#[aztec(noinitcheck)]`:
+
+- The Aztec macro processor skips the [insertion of the initialization check](#initializer-functions-aztecinitializer) for this specific function
+- The function can be called at any time, even if the contract hasn't been initialized yet
+
+## `Internal` functions #[aztec(internal)]
+
+This macro inserts a check at the beginning of the function to ensure that the caller is the contract itself. This is done by adding the following assertion:
+
+```rust
+assert(context.msg_sender() == context.this_address(), "Function can only be called internally");
+```
+
+## Custom notes #[aztec(note)]
+
+Certainly! Let's draft the documentation for the "Custom notes #[aztec(note)]" section, focusing on the functionality provided by this attribute. I'll use the code you've shared to provide accurate and detailed information about how this attribute works under the hood.
```

Comment:
This feels like leftovers from a previous discussion 😄

Reply:
hahaha WOOPS